### PR TITLE
Update/refactor and changelog

### PR DESCRIPTION
**Pull request by ts-leojoseph-r**
# Description
* Created USERGUIDE.md and moved Configuration, Usage, etc., from Readme
* Link Items in changelog to userguide
* Few refactoring in SDK code

## Links
[MINI-1331](https://jira.rakuten-it.com/jira/browse/MINI-1331)

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,36 +20,12 @@ All the MiniApp files downloaded by the MiniApp iOS library are cached locally
```
Comment (corycaywood) [inline README.md:20]: You also need to update this line in `.travis.yml` to point to `USERGUIDE.md`:
https://github.com/rakutentech/ios-miniapp/blob/29f696b2920d23af77da0a31ee0dfc5c9b976cc3/.travis.yml#L45

File: USERGUIDE.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
+# Requirements
```
Comment (corycaywood) [inline USERGUIDE.md:1]: I think the `# MiniApp` and `## Features` sections from the readme need to be moved here. The published Github pages site needs to be able to stand on it's own as an entry point to the SDK documentation: 
https://rakutentech.github.io/ios-miniapp/

Maybe we can keep the `# MiniApp` section in both `README.md` and `USERGUIDE.md`?

File: USERGUIDE.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+# Configuration
+
+In your project configuration .plist you should add below Key/Value :
+
+| Key                          | Type    | Description                                                     | Optional| Default |
+| :----                        | :----:  | :----                                                           |:----:   |:----:   |
+| RASApplicationIdentifier     | String  | `Set your MiniApp host application identifier`                  |NO       |`none`   |
+| RASProjectSubscriptionKey    | String  | `Set your MiniApp subscription key`                             |NO       |`none`   |
+| RMAAPIEndpoint               | String  | `Provide your own Base URL for API requests`                    |NO       |`none`   |
+| RMAIsTestMode                | String  | `Loading mode of the API (`true`: testing, `false`: published)` |NO       |false    |
```
Comment (corycaywood) [inline USERGUIDE.md:24]: `RMAIsTestMode` needs to be removed from the documentation until it's available on the backend. Also, the `Optional` column for this should be `YES`, shouldn't it?

File: README.md (hunk 1 above)
Comment (ts-leojoseph-r) [reply README.md:20]: Done. Updated

File: USERGUIDE.md (hunk 2 above)
Comment (ts-leojoseph-r) [reply USERGUIDE.md]: Agreed. I copied `# MiniApp` and `## Features` sections in USERGUIDE.md.

But sometimes we will end up updating in two places now. README.md and USERGUIDE.md. Guess we may need to come up with some other content for just the Readme.

File: USERGUIDE.md (hunk 3 above)
Comment (ts-leojoseph-r) [reply USERGUIDE.md]: Removed `RMAIsTestMode` details from the documentation. And yes, the default value should be `true`

**Pull request merged**